src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -135,4 +103,77 @@ public long getRetryInterval()
 
   }
 
+  private static ServiceUnavailableRetryStrategy getServiceUnavailableRetryStrategy()
+  {
+    return new ServiceUnavailableRetryStrategy() {
+      private int executionCount = 0;
+      int REQUEST_TIMEOUT = 408;
+
+      @Override
+      public boolean retryRequest(
+              final HttpResponse response, final int executionCount,
+              final HttpContext context)
+      {
+        this.executionCount = executionCount;
+        int statusCode = response.getStatusLine().getStatusCode();
+        LOGGER.info("In retryRequest for service unavailability with statusCode:{} and uri:{}",
+                    statusCode, getRequestUriFromContext(context));
+        boolean needNextRetry = (statusCode == REQUEST_TIMEOUT || statusCode >= 500)
```

Comment:
Yupp makes sense, though will explicitly defining the 5xx which we want to retry now be considered a breaking change? It was one of the reasons I didnt change the logic that was written here. Just refactored to make it more readable.

Reply:
Well, if you are concerned about breaking change, perhaps this can be deferred to a larger design discussion on whether this retry policy does in fact improve availability, or overloads downstream systems and adds latency, without improving availability.  I don't know whether this entire class adds value or causes damage.